src/APICore.ts

```diff
@@ -85,4 +87,19 @@ export class APICore {
   private isThrottled(res: Response): boolean {
     return res.status === 429;
   }
+
+  private validateRetryOptions() {
+    let waitTime = this.options.retryAfter;
+    let totalMs = this.options.retryAfter;
+
+    // Compute the theoretical maximum execution time to ensure it's less than 1.5 hours.
+    for (let i = 1; i < this.options.maxRetries; i++) {
+      waitTime *= this.options.timeMultiple;
+      totalMs += waitTime;
+    }
+
+    if (totalMs > 5400000) {
```

Comment:
This should be configurable(an option).

Or (better) instead of adding this cap here, change the default `retryAfter/maxRetries/timeMultiple` so that it approximately end up equal to 1.5hours

Reply:
> This should be configurable(an option).
I'm surprised by this suggestion. I thought the idea behind enforcing a maximum time limit is to essentially save the implementer (and their org) from themselves 😅 If we make this configurable, we might as well just not check for this at all, right?

> change the default retryAfter/maxRetries/timeMultiple so that it approximately end up equal to 1.5hours
This is actually already the case 🙂 I used the benchmarks(?) provided by Pascal to set the default values, and with the current setup, it will take up to 1 hour, 25 minutes, and 15 seconds.

Reply:
I think the default should be sane (what you've done here), but I think it should still be possible for someone to just configure "retry until the end of time" if they explicitly request it/configure it that way.

1-Maybe retrying for 3 hours is actually legit 🤷, it's hard to tell what is a good "upper limit" that is not arbitrary.
2- Coveo platform would not suffer from this, I'm pretty certain.
3- Could instead just log warnings when we see some ridiculous value (ie: days worth of retry or something like that), but still accept it and not throw.

@pdoyon-coveo : Any opinion ?

Reply:
I agree with that. A good documentation, default values that make sense in most cases, and I would not enforce any limit because we don't know all use cases out there. If a client needs to retry once per day, then it should be possible to do so.

As discussed, it would have been nice to also have another option that would be the maximum time to wait per retry to protect the client against himself. But it appears to be more complicated to implement, so let's go without this option.

Reply:
Alrighty, I've removed the check so implementers are no longer forced to keeping their configuration to under 1.5 hours and updated the docs accordingly ✅